app/helpers/application_helper/button/basic.rb

```diff
@@ -12,6 +12,11 @@ def initialize(view_context, view_binding, instance_data, props)
     end
   end
 
+  def check_rbac?
+    return true if self[:child_id].nil?
+    role_allows?(:feature => self[:child_id])
```

Comment:
What is a `child_id`?  Is that a feature identifier?

Note, we'll have to change this for multiple entitlements to pass in the "context", namely the resource you're asking about.  Basically, all `roles_allows?` calls will have to ask am I allowed to do "this feature" for this "resource" (vm/template/service/host/etc.).